packages/publint/tests/playground.test.js

```diff
@@ -201,35 +199,41 @@ testFixture('invalid-repository-value-object-deprecated', [
  */
 function testFixture(name, expectCodes, options) {
   test(name, { concurrent: true }, async ({ expect }) => {
-    const fixtureName = name.replace(/\(.*$/, '').trim()
-    const pkgDir = path.resolve(process.cwd(), 'tests/fixtures', fixtureName)
-    const { messages } = await publint({
-      pkgDir,
-      level: options?.level,
-      strict: options?.strict,
-    })
-
-    // unfortunately the messages are not always in order as checks are ran in parallel,
-    // here we sort it to make the tests more consistent
-    messages.sort((a, b) => a.code.localeCompare(b.code))
-
-    if (options?.debug) {
-      const pkg = JSON.parse(
-        await fs.readFile(path.join(pkgDir, 'package.json'), 'utf-8'),
-      )
-      console.log()
-      console.log('Logs:', name)
-      messages.forEach((m) => console.log(formatMessage(m, pkg)))
-      console.log()
-    }
-
-    // you can test an array of objects
-    if (typeof expectCodes[0] === 'object') {
-      const codes = messages.map((v) => ({ code: v.code, type: v.type }))
-      expect(codes).toEqual(expectCodes)
-    } else {
-      const codes = messages.map((v) => v.code)
-      expect(codes).toEqual(expectCodes)
+    const fixtureName = name.replace(/\(.*$/, '').trim() + '.js'
+    const fixtureDir = path.resolve(process.cwd(), 'tests/fixtures')
+    const fixturePath = path.resolve(fixtureDir, fixtureName)
+    const fixtureContent = (await import(fixturePath)).default
+    const fixture = await createFixture(fixtureContent, { tempDir: fixtureDir })
```

Comment:
Setting `tempDir` when creating the fixture seems to have fixed the flakiness I was experiencing with the Windows tests.

Reply:
I think I'd lean on setting the timeout still to keep the project files cleaner. There's also a timeout set at https://github.com/publint/publint/blob/1af86c22aea783a2f697911e5a58d485f28118f7/packages/pack/tests/pack-as-list.test.js#L98 so it'd be a bit consistent for now.

Reply:
I did originally have the timeout set but the Windows tests were still failing occasionally even with a higher timeout of 8s. Here's an example of it failing with the timeout set to 10s: https://github.com/publint/publint/actions/runs/12869704032/job/35879125225?pr=132

Setting `tempDir` was the only thing that fixed it for me. I can choose a different location if that works better.